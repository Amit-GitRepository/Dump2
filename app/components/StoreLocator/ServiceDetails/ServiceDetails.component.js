/* ServiceDetails: show static details of each service selected from StoreDetails*/

import PropTypes from 'prop-types';
import React, {Component} from 'react';
import result from 'lodash/result';
import styles from './ServiceDetails.style';
import {Card, List} from '../../Shared';
import {View} from 'react-native';

class ServiceDetails extends Component {
  render () {
    const {navigation} = this.props;
    const serviceDetails = result(navigation, 'state.params', {});
    return (
      <View style={styles.container}>
        <View style={styles.shadowContainer} elevation={5}>
          <Card header={serviceDetails.serviceName + ' Service'} cardWrapperStyles={styles.cardWrapperStyles} elevation={0}>
            <List data={serviceDetails.products} />
          </Card>
        </View>
      </View>);
  }
}
ServiceDetails.defaultProps = {
  navigation: {}
};
ServiceDetails.propTypes = {
  navigation: PropTypes.object
};
export default ServiceDetails;
