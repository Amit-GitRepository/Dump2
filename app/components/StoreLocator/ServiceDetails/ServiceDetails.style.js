import {shadowScattered} from '../../../themes/application.styles';

export default {
  container: {
    flex: 1,
    margin: 10
  },
  cardWrapperStyles: {
    marginBottom: -15
  },
  shadowContainer: {
    flex: 1,
    borderRadius: 8,
    ...shadowScattered
  }
};
