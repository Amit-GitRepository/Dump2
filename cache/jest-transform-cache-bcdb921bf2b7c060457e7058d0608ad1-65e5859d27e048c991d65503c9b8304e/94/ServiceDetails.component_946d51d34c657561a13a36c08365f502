Object.defineProperty(exports,"__esModule",{value:true});var cov_1y4jkhprhd=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/StoreLocator/ServiceDetails/ServiceDetails.component.js',hash='71e5b0b1b63317b52ae9b438a34ac9c01fc823da',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/StoreLocator/ServiceDetails/ServiceDetails.component.js',statementMap:{'0':{start:{line:12,column:25},end:{line:12,column:35}},'1':{start:{line:13,column:27},end:{line:13,column:65}},'2':{start:{line:14,column:4},end:{line:19,column:15}},'3':{start:{line:22,column:0},end:{line:24,column:2}},'4':{start:{line:25,column:0},end:{line:27,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:11,column:2},end:{line:11,column:3}},loc:{start:{line:11,column:12},end:{line:20,column:3}},line:11}},branchMap:{},s:{'0':0,'1':0,'2':0,'3':0,'4':0},f:{'0':0},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();

var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _result=require('lodash/result');var _result2=_interopRequireDefault(_result);
var _ServiceDetails=require('./ServiceDetails.style');var _ServiceDetails2=_interopRequireDefault(_ServiceDetails);
var _Shared=require('../../Shared');
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

ServiceDetails=function(_Component){_inherits(ServiceDetails,_Component);function ServiceDetails(){_classCallCheck(this,ServiceDetails);return _possibleConstructorReturn(this,(ServiceDetails.__proto__||Object.getPrototypeOf(ServiceDetails)).apply(this,arguments));}_createClass(ServiceDetails,[{key:'render',value:function render()
{cov_1y4jkhprhd.f[0]++;var _ref=(cov_1y4jkhprhd.s[0]++,
this.props),navigation=_ref.navigation;
var serviceDetails=(cov_1y4jkhprhd.s[1]++,(0,_result2.default)(navigation,'state.params',{}));cov_1y4jkhprhd.s[2]++;
return(
_react2.default.createElement(_reactNative.View,{style:_ServiceDetails2.default.container},
_react2.default.createElement(_Shared.Card,{header:serviceDetails.serviceName+' Service'},
_react2.default.createElement(_Shared.List,{data:serviceDetails.products}))));


}}]);return ServiceDetails;}(_react.Component);cov_1y4jkhprhd.s[3]++;

ServiceDetails.defaultProps={
navigation:{}};cov_1y4jkhprhd.s[4]++;

ServiceDetails.propTypes={
navigation:_propTypes2.default.object};exports.default=

ServiceDetails;