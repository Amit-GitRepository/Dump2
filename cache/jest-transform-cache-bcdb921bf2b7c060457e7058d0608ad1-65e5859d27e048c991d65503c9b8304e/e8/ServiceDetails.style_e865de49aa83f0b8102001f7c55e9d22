Object.defineProperty(exports,"__esModule",{value:true});var cov_1yeg85pph=function(){var path="/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/StoreLocator/ServiceDetails/ServiceDetails.style.js",hash="7b78eb1b5b28eeaf39360fc8505b0a539aa2b334",global=new Function('return this')(),gcv="__coverage__",coverageData={path:"/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/StoreLocator/ServiceDetails/ServiceDetails.style.js",statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:"332fd63041d2c1bcb487cc26dd0d5f7d97098a6c"},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();exports.default={
container:{
padding:10,
flex:1}};