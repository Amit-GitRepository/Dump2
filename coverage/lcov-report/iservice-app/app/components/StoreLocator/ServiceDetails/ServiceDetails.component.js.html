<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for iservice-app/app/components/StoreLocator/ServiceDetails/ServiceDetails.component.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">iservice-app/app/components/StoreLocator/ServiceDetails</a> ServiceDetails.component.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">38x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">38x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">/* ServiceDetails: show static details of each service selected from StoreDetails*/
&nbsp;
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import result from 'lodash/result';
import styles from './ServiceDetails.style';
import {Card, List} from '../../Shared';
import {View} from 'react-native';
&nbsp;
class ServiceDetails extends Component {
  render () {
    const {navigation} = this.props;
    const serviceDetails = result(navigation, 'state.params', {});
    return (
      &lt;View style={styles.container}&gt;
        &lt;Card header={serviceDetails.serviceName + ' Service'}&gt;
          &lt;List data={serviceDetails.products} /&gt;
        &lt;/Card&gt;
      &lt;/View&gt;);
  }
}
ServiceDetails.defaultProps = {
  navigation: {}
};
ServiceDetails.propTypes = {
  navigation: PropTypes.object
};
export default ServiceDetails;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 03 2017 16:14:06 GMT+0530 (IST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
