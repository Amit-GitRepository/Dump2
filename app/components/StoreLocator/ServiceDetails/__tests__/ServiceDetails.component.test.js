import React from 'react';
import renderer from 'react-test-renderer';
import ServiceDetails from '../ServiceDetails.component';
import set from 'lodash/set';

describe('ServiceDetails component', () => {
  it('ServiceDetails: renders correctly', () => {
    const navigation = {};
    set(navigation, 'state.params.products', ['SomeProductName']);
    set(navigation, 'state.params.serviceName', 'SomeserviceName');
    const tree = renderer.create(<ServiceDetails navigation={navigation}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
