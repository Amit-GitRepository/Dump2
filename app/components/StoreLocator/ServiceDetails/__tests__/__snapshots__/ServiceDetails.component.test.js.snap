// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceDetails component ServiceDetails: renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "margin": 10,
    }
  }
>
  <View
    elevation={5}
    style={
      Object {
        "borderRadius": 8,
        "flex": 1,
        "shadowColor": "#303C46",
        "shadowOffset": Object {
          "height": 2,
          "width": 1,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 2,
      }
    }
  >
    <View
      elevation={0}
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 8,
            "flex": 1,
            "marginHorizontal": -1,
            "overflow": "hidden",
            "shadowColor": "#303C46",
            "shadowOffset": Object {
              "height": 2,
              "width": 1,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 1,
          },
          Object {},
        ]
      }
    >
      <View
        onPress={null}
        style={
          Array [
            Object {
              "zIndex": 1,
            },
            Object {
              "marginBottom": -15,
            },
          ]
        }
      >
        <View>
          <View
            style={
              Object {
                "borderWidth": 0,
                "elevation": 3,
                "shadowColor": "#303C46",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 1,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderBottomColor": "#F93E60",
                    "paddingHorizontal": 5,
                  },
                  Object {
                    "borderBottomColor": "#303C46",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#E9E9EA",
                        "padding": 10,
                      },
                      Object {
                        "backgroundColor": "#303C46",
                        "marginHorizontal": -5,
                        "padding": 10,
                      },
                    ],
                    Object {},
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#303C46",
                      },
                      Array [
                        Object {
                          "color": "#000000",
                          "fontSize": 18,
                        },
                        Object {
                          "color": "#FFFFFF",
                        },
                      ],
                      Object {
                        "fontFamily": "SukhumvitSet-Bold",
                      },
                    ]
                  }
                >
                  SomeserviceName Service
                </Text>
              </View>
            </View>
          </View>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#303C46",
                  "fontSize": 24,
                },
                Array [
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "bottom": 9.6,
                    "marginBottom": -9.6,
                  },
                  Object {},
                ],
                Object {
                  "fontFamily": "iservice",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            Ó§Å
          </Text>
        </View>
      </View>
      <RCTScrollView
        data={
          Array [
            "SomeProductName",
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              id={0}
              onPress={null}
            >
              <View
                animation="fadeIn"
                duration={500}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "paddingHorizontal": 10,
                      "paddingVertical": 15,
                    },
                    Object {
                      "backgroundColor": "#FFFFFF",
                    },
                  ]
                }
              >
                <View
                  style={Object {}}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#303C46",
                        "fontFamily": "SukhumvitSet-SemiBold",
                        "fontSize": 18,
                        "marginBottom": -4,
                      }
                    }
                  >
                    SomeProductName
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
